cmake.deps/cmake/BuildTreesitterParsers.cmake

```diff
@@ -1,29 +1,32 @@
-ExternalProject_Add(treesitter-c
-PREFIX ${DEPS_BUILD_DIR}
-URL ${TREESITTER_C_URL}
-DOWNLOAD_DIR ${DEPS_DOWNLOAD_DIR}/treesitter-c
-DOWNLOAD_COMMAND ${CMAKE_COMMAND}
-  -DPREFIX=${DEPS_BUILD_DIR}
-  -DDOWNLOAD_DIR=${DEPS_DOWNLOAD_DIR}/treesitter-c
-  -DURL=${TREESITTER_C_URL}
-  -DEXPECTED_SHA256=${TREESITTER_C_SHA256}
-  -DTARGET=treesitter-c
-  -DUSE_EXISTING_SRC_DIR=${USE_EXISTING_SRC_DIR}
-  -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/DownloadAndExtractFile.cmake
-PATCH_COMMAND ${CMAKE_COMMAND} -E copy
-  ${CMAKE_CURRENT_SOURCE_DIR}/cmake/TreesitterParserCMakeLists.txt
-        ${DEPS_BUILD_DIR}/src/treesitter-c/CMakeLists.txt
-CMAKE_ARGS
-  -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
-  -DCMAKE_GENERATOR=${CMAKE_GENERATOR}
-  -DCMAKE_GENERATOR_PLATFORM=${CMAKE_GENERATOR_PLATFORM}
-  ${BUILD_TYPE_STRING}
-  -DCMAKE_INSTALL_PREFIX=${DEPS_INSTALL_DIR}
-  -DCMAKE_OSX_ARCHITECTURES=${CMAKE_OSX_ARCHITECTURES_ALT_SEP}
-  # Pass toolchain
-  -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}
-  -DPARSERLANG=c
+function(BuildTSParser LANG TS_URL TS_SHA256 TS_CMAKE_FILE)
+  set(NAME treesitter-${LANG})
+  ExternalProject_Add(${NAME}
+  PREFIX ${DEPS_BUILD_DIR}
+  URL ${TREESITTER_C_URL}
+  DOWNLOAD_DIR ${DEPS_DOWNLOAD_DIR}/${NAME}
+  DOWNLOAD_COMMAND ${CMAKE_COMMAND}
+    -DPREFIX=${DEPS_BUILD_DIR}
+    -DDOWNLOAD_DIR=${DEPS_DOWNLOAD_DIR}/${NAME}
+    -DURL=${TS_URL}
+    -DEXPECTED_SHA256=${TS_SHA256}
+    -DTARGET=${NAME}
+    -DUSE_EXISTING_SRC_DIR=${USE_EXISTING_SRC_DIR}
+    -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/DownloadAndExtractFile.cmake
+  PATCH_COMMAND ${CMAKE_COMMAND} -E copy
+      ${CMAKE_CURRENT_SOURCE_DIR}/cmake/${TS_CMAKE_FILE}
+      ${DEPS_BUILD_DIR}/src/${NAME}/CMakeLists.txt
+  CMAKE_ARGS
+    -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+    -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
```

Comment:
This variable is unused. Delete.

Reply:
Is that still the case when building a  `scanner.cc`?

Reply:
I don't think `CMAKE_ARGS` are necessary since the defaults are not changed

Reply:
nor is the patch command?

Reply:
> Is that still the case when building a scanner.cc?

Unsure. I will test.

Reply:
The Treesittercxx doesn't work properly. Let's remove `CMAKE_CXX_COMPILER` for now since it gives a warning and deal with it later if we need a C++ parser.

Reply:
Anything else I should remove from that file? Everything but `-DPARSERLANG` should be default.

Reply:
> nor is the patch command?

Nah, the "patching" is necessary (technically not a patch, but it's the most convenient way to get cmake to work.)

You can remove all arguments from `CMAKE_ARGS` (including the toolchain comment) except for `-DPARSERLANG` and 
`-DCMAKE_INSTALL_PREFIX`. Default for `CMAKE_INSTALL_PREFIX` is `/usr/local`.